### DISCUSSION TITLE
Crop tool for badge image upload

### DISCUSSION THREAD

**Discussion by Thor368** [pretix/pretix#3014] [Ideas]
Hello!
I think it would be a nice addition to implement a crop tool for the badge image upload. Many people dont have the technical capabilities or skill to do that on their own. Especially on mobile.

**Comment by raphaelm** [comment]
This exists, for a specific aspect ratio of image (3:4). You can enable it in the advanced settings of the upload question.

**Reply by Thor368** [reply]
That is a nice feature in deed! What I was reffering to was a "pan/zoom" cropping feature tho.

**Reply by raphaelm** [reply]
I would describe the existing feature as capable of that, if I understand you correctly. Or can you show an example from some other app to visualize what you mean?

**Reply by Thor368** [reply]
OK! This option does way more than I thought it would do xO Only two things left then:
1. Make the aspect ratio configurable.
2. Maaaaybe make the feature name more clear in the backend xD

**Reply by raphaelm** [reply]
1. not planned currently, sorry, to limit complexity, as this also affects taking pictures in all our apps (pretixSCAN on three platforms, pretixPOS)
2. happy for suggestions!

**Reply by Thor368** [reply]
Ah, that's too bad. OK, I'd call it something like "Enable croptool (3:4 aspect enforced)". German translation "Croptool aktivieren (erzwingt 3:4 Seitenverhältnis)"

**Reply by raphaelm** [reply]
Hmm, no, that's on purpose that we didn't name it like that.

The feature is mostly a semantical "this is a picture of a person" and the crop tool is a means to get there. The checkbox might also have additional effects.

But there's probably some middle ground.

**Reply by Thor368** [reply]
The question type is "File-upload" which could litteraly be anything or any picture type. The integrated crop tool is realy powerfull. Limiting it to only "this is a picture of a person" might be a less than ideal solution. As I said: Great feature but could use some touch-ups :)

I understand that some code further down the stream might read that option and say "hey, i assume a 3:4 image here" and react unpredictable with anything else. My suggestion for a clean solution is adding an option in the file-upload question type parallel to the portrait option, enabling the crop tool with a free or even better configurable aspect ratio.

BTW. I identifyed the section where the aspect ratio is fixed:
https://github.com/pretix/pretix/blob/master/src/pretix/static/pretixpresale/js/ui/questions.js#L94

For freeing that aspect ratio simply set it to NaN. I'd actually try that but I'm not yet familiar with the build process of pretix.
